microsoft/testsuites/virtualclient/virtualclient.py

```diff
@@ -0,0 +1,133 @@
+import re
+import uuid
+from functools import partial
+from typing import Any, Dict, List, Tuple, cast
+
+from assertpy import assert_that
+
+from lisa import (
+    Logger,
+    RemoteNode,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.environment import Node
+from lisa.features import Disk
+from lisa.operating_system import Debian
+from lisa.testsuite import TestResult
+from lisa.tools import VirtualClientTool
+from lisa.util.parallel import run_in_parallel
+
+
+@TestSuiteMetadata(
+    area="virtual_client",
+    category="performance",
+    description="""
+        This test suite runs the performance test cases with Virtual Client.
+    """,
+)
+class VirtualClient(TestSuite):
+    # [07/08/2024 13:58:15] Exit Code: 0\r\n
+    exit_code_pattern = re.compile(r"([\w\W]*?)Exit Code: (?P<exit_code>\d+)")
+
+    @TestCaseMetadata(
+        description="""
+            This test is to run redis workload testing with Virtual Client.
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            supported_os=[Debian],
+            min_count=2,
+        ),
+    )
+    def perf_vc_redis(self, log: Logger, result: TestResult) -> None:
+        self._run_work_load(result=result, profile_name="PERF-REDIS")
+
+    @TestCaseMetadata(
+        description="""
+            This test is to run PostgreSQL workload testing with Virtual Client.
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            supported_os=[Debian],
+            min_count=2,
+        ),
+        timeout=3000,
+    )
+    def perf_vc_postgresql(self, log: Logger, result: TestResult) -> None:
+        environment = result.environment
+        assert environment, "the environment should not be empty"
+        for node in environment.nodes.list():
+            disk = node.features[Disk]
+            disk.add_data_disk(count=1, size_in_gb=256)
+
+        self._run_work_load(
+            result=result,
+            profile_name="PERF-POSTGRESQL-HAMMERDB-TPCC",
+            timeout=45,
+        )
+
+    def _run_work_load(
+        self,
+        result: TestResult,
+        profile_name: str,
+        system: str = "Azure",
+        timeout: int = 10,
+    ) -> None:
+        environment = result.environment
+        assert environment, "the environment of test result shouldn't be empty"
+
+        nodes: List[Dict[str, Any]] = [
```

Comment:
Use a class to present the data structure, instead of a dict. It's easier to use and check name errors.

Reply:
updated